src/stdlib/SDL_string.c

```diff
@@ -1941,4 +1941,60 @@ SDL_vsnprintf(SDL_OUT_Z_CAP(maxlen) char *text, size_t maxlen, const char *fmt,
 }
 #endif /* HAVE_VSNPRINTF */
 
+int
+SDL_asprintf(char **strp, SDL_PRINTF_FORMAT_STRING const char *fmt, ...)
+{
+    va_list ap;
+    int retval;
+
+    va_start(ap, fmt);
+    retval = SDL_vasprintf(strp, fmt, ap);
+    va_end(ap);
+
+    return retval;
+}
+
+int
+SDL_vasprintf(char **strp, const char *fmt, va_list ap)
+{
+    int retval;
+    int size = 100;     /* Guess we need no more than 100 bytes */
```

Comment:
This may simplify the implementation, but this restriction here is not a good one at all.

Reply:
OK, I take it back - you _are_ `realloc`'ing.

Reply:
If it the resulting string is larger than that, the buffer will be reallocated based on the return value from `SDL_snprintf` and it'll try again with the larger buffer size. Detecting the length without an initial buffer requires `snprintf` to report the size when the first parameter is `NULL`, which is in C99 but not SUSv2.

Having said that, though, SDL's built-in version of `snprintf` doesn't seem to be compliant with either standard, since it doesn't return the full length of the string if the output is truncated. As such, this PR can't be merged until that issue is resolved.